sockeye/extract.py

```diff
@@ -0,0 +1,112 @@
+import numpy as np
+
+from sockeye.log import setup_main_logger, log_sockeye_version
+from . import arguments
+from . import constants as C
+from . import utils
+
+logger = setup_main_logger(__name__, console=True, file_logging=False)
+
+def _extract(param_names: List[str],
+             params: Dict[str, mx.nd.NDArray],
+             ext_params: Dict[str, np.ndarray]) -> List[str]:
+    """
+    Extract specific parameters from a given base.
+
+    :param param_names: Names of parameters to be extracted.
+    :param params: Mapping from parameter names to the actual NDArrays parameters.
+    :param ext_params: Extracted parameter dictionary.
+    :return: Remaining names of parameters to be extracted.
+    """
+    remaining_param_names = list(param_names)
+    for name in param_names:
+        if name in params:
+            logger.info("\tFound '%s'", name)
+            ext_params[name] = params[name].asnumpy() 
+            remaining_param_names.remove(name)
+    return remaining_param_names
+
+
+def extract(param_path: str,
+            param_names: List[str],
+            list_all: bool) -> Dict[str, np.ndarray]:
+    """
+    Extract specific parameters given their names.
+
+    :param param_path: Path to the parameter file.
+    :param param_names: Names of parameters to be extracted.
+    :param list_all: List names of all available parameters.
+    :return: Extracted parameter dictionary.
+    """
+    logger.info("Loading parameters from '%s'", param_path)
+    arg_params, aux_params = utils.load_params(param_path)
+
+    ext_params = {} # type: Dict[str, np.ndarray]
+    param_names = _extract(param_names, arg_params, ext_params)
+    param_names = _extract(param_names, aux_params, ext_params)
+
+    if len(param_names) > 0:
+        logger.info("The following parameters were not found:")
+        for name in param_names:
+            logger.info("\t%s", name)
+        logger.info("Check the following availabilities")
+        list_all = True
+
+    if list_all:
+        if arg_params:
+            logger.info("Available arg parameters:")
+            for name in arg_params:
+                logger.info("\t%s: shape=%s", name, str(arg_params[name].shape))
+        if aux_params:
+            logger.info("Available aux parameters:")
+            for name in aux_params:
+                logger.info("\t%s: shape=%s", name, str(aux_params[name].shape))
+
+    return ext_params
+
+
+def main():
+    """
+    Commandline interface to extract parameters.
+    """
+    log_sockeye_version(logger)
+    params = argparse.ArgumentParser(description="Extract specific parameters.")
+    arguments.add_extract_args(params)
+    args = params.parse_args()
+
+    if os.path.isdir(args.input):
+        param_paths = os.path.join(args.input, C.PARAMS_BEST_NAME)
+    else:
+        param_paths = args.input
+    ext_params = extract(param_paths, args.names, args.list_all)
+    
+    if len(ext_params) > 0:
+        assert args.output != None
```

Comment:
could you replace this with  `utils.check_condition` and provide a user-friendly error message?